msdk-featdet/msdk-featdet-openms/src/main/java/io/github/msdk/featdet/openms/FeatureFinderMetaboDetector.java

```diff
@@ -0,0 +1,108 @@
+/* 
+ * (C) Copyright 2015-2016 by MSDK Development Team
+ *
+ * This software is dual-licensed under either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+
+package io.github.msdk.featdet.openms;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.annotation.Nonnull;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+public class FeatureFinderMetaboDetector {
+	private final String OPENMS_FEATURE_FINDER_METABO_LIBRARY_NAME;
+	private final File FEATURE_XML_FILE;
+
+	private final @Nonnull File mzMLFile;
+
+	public FeatureFinderMetaboDetector(String mzMLFileName) throws FeatureFinderMetaboException {
+		this(new File(mzMLFileName));
+	}
+
+	public FeatureFinderMetaboDetector(File mzMLFile) throws FeatureFinderMetaboException {
+		this.mzMLFile = mzMLFile;
+		FEATURE_XML_FILE = new File(mzMLFile.getName() + "featureXML");
+		OPENMS_FEATURE_FINDER_METABO_LIBRARY_NAME = FeatureFinderMetaboLocator.findFeatureFinderMetabo();
+	}
+
+	public List<Double> execute()
+			throws SAXException, IOException, ParserConfigurationException, FeatureFinderMetaboException {
+		List<Double> mzValuesResult = new ArrayList<Double>();
+
+		if (!OPENMS_FEATURE_FINDER_METABO_LIBRARY_NAME.equals(null)) {
```

Comment:
Instead of checking !OPENMS_FEATURE_FINDER_METABO_LIBRARY_NAME, you can check for OPENMS_FEATURE_FINDER_METABO_LIBRARY_NAME and fail quickly. This way you avoid long nested _if_ clauses.